js/widgets/popup.js

```diff
@@ -865,14 +857,16 @@ $.widget( "mobile.popup", {
 			// If that happens and we remove the container a few lines below, we
 			// will cause an infinite recursion - #5244
 			.detach()
-			.insertAfter( this._ui.placeholder )
-			.removeClass( "ui-popup ui-overlay-shadow ui-corner-all ui-body-inherit" );
+			.insertAfter( this._ui.placeholder );
 		this._ui.screen.remove();
 		this._ui.container.remove();
 		this._ui.placeholder.remove();
 	},
 
 	_destroy: function() {
+		if ( this.options.enhanced ) {
+			this.classesElementLookup = {};
+		}
```

Comment:
Of course! Isn't that the point of enhanced? All classes are present in the original markup, and all classes must continue to be present after the widget is destroyed. Sure, ideally we'd restore the widget to its initial state upon ```destroy()``` - that is, if you changed, say, the theme option via ```.option()```, we'd restore the theme upon ```destroy()``` while leaving all the other classes intact, but we've established that that's simply not feasible for all possible state changes a widget can experience during its lifetime, so we simply leave all the DOM elements and classes in place upon ```destroy()```. This accomplishes that.